spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -115,7 +114,7 @@ def call_kernel(self, interrupt=False, blocking=False, callback=None):
     def set_kernel_client_and_manager(self, kernel_client, kernel_manager):
         """Set the kernel client and manager"""
         self.kernel_manager = kernel_manager
-        self.kernel_client = kernel_client
+        self.kernel_client = self.patch_kernel_client(kernel_client)
```

Comment:
Let's leave this as before, and call `redefine_complete_for_dbg` below, like this

```python
self.spyder_kernel_comm.open_comm(kernel_client)

# Redefine the complete method to work while debugging.
self.redefine_complete_for_dbg(self.kernel_client)
```